lib/native/source/opensmt/api.patch

```diff
@@ -0,0 +1,537 @@
++
++}
+diff --git a/src/sorts/SSort.h b/src/sorts/SSort.h
+index 990da3b0..14922397 100644
+--- a/src/sorts/SSort.h
++++ b/src/sorts/SSort.h
+@@ -40,7 +40,7 @@ struct SSymRef {
+     inline friend bool operator!= (SSymRef a1, SSymRef a2) {return a1.x != a2.x; }
+ };
+ 
+-const static struct SSymRef SSymRef_Undef = {INT32_MAX};
++static const struct SSymRef SSymRef_Undef = {INT32_MAX};
+ 
+ struct SSymRefHash {
+     uint32_t operator () (SSymRef s) const {
+@@ -66,8 +66,8 @@ struct SRef {
+     inline friend bool operator!= (SRef a1, SRef a2) {return a1.x != a2.x; }
+ };
+ 
+-const static struct SRef SRef_Undef = {INT32_MAX};
+-const static struct SRef SRef_Nil   = {INT32_MAX-1};
++static const struct SRef SRef_Undef = {INT32_MAX};
++static const struct SRef SRef_Nil   = {INT32_MAX-1};
+ 
+ struct SRefHash {
+     uint32_t operator () (const SRef& s) const {
+diff --git a/src/symbols/SymRef.h b/src/symbols/SymRef.h
+index 693d332f..315cee13 100644
+--- a/src/symbols/SymRef.h
++++ b/src/symbols/SymRef.h
+@@ -35,17 +35,19 @@ struct SymRef {
+     inline friend bool operator!= (const SymRef& a1, const SymRef& a2) {return a1.x != a2.x; }
+ };
+ 
+-const static struct SymRef SymRef_Undef = {INT32_MAX};
+-const static struct SymRef SymRef_Nil   = {INT32_MAX-1};
++static const struct SymRef SymRef_Undef = {INT32_MAX};
++static const struct SymRef SymRef_Nil   = {INT32_MAX-1};
+ 
+ struct SymRefHash {
+     uint32_t operator () (const SymRef& s) const {
+         return (uint32_t)s.x; }
+ };
+ 
++/*
+ template <>
+ struct Equal<const SymRef> {
+     bool operator() (const SymRef& s1, const SymRef& s2) { return s1 == s2; }
+ };
++*/
+ 
+ #endif //OPENSMT_SYMREF_H
+diff --git a/src/symbols/SymStore.cc b/src/symbols/SymStore.cc
+index 9b2c3241..a2b06c2d 100644
+--- a/src/symbols/SymStore.cc
++++ b/src/symbols/SymStore.cc
+@@ -27,6 +27,7 @@ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ #include "string.h"
+ #include "SymStore.h"
+ #include "Symbol.h"
++#include "OsmtApiException.h"
+ 
+ const char* SymStore::e_duplicate_symbol = "Symbol Store: symbol already exists";
+ const int   SymStore::symstore_buf_offs_idx = 1;
+@@ -55,7 +56,12 @@ SymRef SymStore::newSymb(const char * fname, SRef rsort, vec<SRef> const & args,
+                 }
+             }
+         }
++        if (!symConfig.isInterpreted) {
++          // Only builtin functions like "=" or "+" should be polymorphic
++          throw OsmtApiException("Symbol `" + std::string(fname) + "` has conflicting definitions");
++        }
+     }
++
+     bool newsym = (symrefs == nullptr);
+     SymRef tr = ta.alloc(rsort, args, symConfig);
+     SymId id = symbols.size();
+diff --git a/src/symbols/Symbol.h b/src/symbols/Symbol.h
+index e795bd56..4b6f6018 100644
+--- a/src/symbols/Symbol.h
++++ b/src/symbols/Symbol.h
+@@ -48,20 +48,20 @@ struct SymbolConfig {
+ };
+ 
+ namespace SymConf {
+-    constexpr auto Default = SymbolConfig{false, false, false, SymbolProperty::None};
```

Comment:
Why do we need to remove "auto" as modifier? Does SWIG not support "auto"?

Reply:
I think this was causing parsing issues with SWIG as well, but I'll have to go back and check.